### PR TITLE
ZK-4312: leaking request/response into wpd cache

### PR DESCRIPTION
**Pull request by plijyfes**

### CODE REVIEW

File: zk/src/org/zkoss/zk/ui/http/AbstractExtendlet.java [hunk 1]
```diff
@@ -142,6 +142,8 @@ else if (ServletContext.class.isAssignableFrom(argTypes[j]))
 		} catch (Throwable ex) { //log and eat ex
 			log.error("Unable to invoke " + mi.method, ex);
 			return "";
+		} finally {
+			mi.arguments = new Object[0];
```
[the remark this answers is not in the document]
Comment (DevChu) [reply zk/src/org/zkoss/zk/ui/http/AbstractExtendlet.java]: Arrays.fill(mi.arguments, null);

File: zk/src/org/zkoss/zk/ui/http/AbstractExtendlet.java [hunk 2]
```diff
@@ -142,6 +143,8 @@ else if (ServletContext.class.isAssignableFrom(argTypes[j]))
 		} catch (Throwable ex) { //log and eat ex
 			log.error("Unable to invoke " + mi.method, ex);
 			return "";
+		} finally {
+			Arrays.fill(mi.arguments, null);
```
Comment (plijyfes) [inline zk/src/org/zkoss/zk/ui/http/AbstractExtendlet.java]: comment

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**